### PR TITLE
[Layers] Add basic operation layers

### PR DESCRIPTION
**Pull request by baek2sm**
[Done]
- added subtraction layer
- added multiplication layer
- added division layer
- added sementic unittest for above layers

[Todo]
I'll update the existing functions to be compatible with the latest version(2.17) of tensorflow, so I'll upload the golden data unittest as a separate commit.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by jijoongmoon** [comment]
> I'll update the existing functions to be compatible with the latest version(2.17) of tensorflow, so I'll upload the golden data unittest as a separate commit.

How about using torch?

File: nntrainer/layers/multiplication_layer.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+static constexpr size_t SINGLE_INOUT_IDX = 0;
+
+void MultiplicationLayer::finalize(InitLayerContext &context) {
+  context.setOutputDimensions({context.getInputDimensions()[0]});
+}
+
+void MultiplicationLayer::forwarding(RunLayerContext &context, bool training) {
+  Tensor &hidden_ = context.getOutput(SINGLE_INOUT_IDX);
+
+  for (unsigned int idx = 0; idx < context.getNumInputs(); ++idx) {
```
Comment (jijoongmoon) [inline nntrainer/layers/multiplication_layer.cpp:33]: Is it supposed to have binary input?? If it is, we do not need to copy. just get the two input and multiply.  Then we can save more latency.

File: nntrainer/layers/multiplication_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        hidden_step.copy(input_step);
+      } else {
+        hidden_step.multiply_i(input_step);
+      }
+    }
+  }
+}
+
+void MultiplicationLayer::calcDerivative(RunLayerContext &context) {
+  context.getOutgoingDerivative(0).copy(
```
Comment (jijoongmoon) [inline nntrainer/layers/multiplication_layer.cpp:85]: I don't think we need copy here. just do the computation and save it to output Tensor.

**Review by skykongkong8 — APPROVED** [review]
This PR would be a great guide for anyone who wants to implement a new layer in the nntrainer.

**Comment by baek2sm** [comment]
I will update scripts and push PR again. Thanks.

File: api/nntrainer-api-common.h [hunk 3]
```diff
@@ -62,8 +62,12 @@ typedef enum {
     27, /**< Layer Normalization Layer type (Since 7.0) */
   ML_TRAIN_LAYER_TYPE_POSITIONAL_ENCODING =
     28, /**< Positional Encoding Layer type (Since 7.0) */
-  ML_TRAIN_LAYER_TYPE_IDENTITY = 29, /**< Identity Layer type (Since 8.0) */
-  ML_TRAIN_LAYER_TYPE_SWIGLU = 30,   /**< Swiglu Layer type */
+  ML_TRAIN_LAYER_TYPE_IDENTITY = 29,    /**< Identity Layer type (Since 8.0) */
+  ML_TRAIN_LAYER_TYPE_SWIGLU = 30,      /**< Swiglu Layer type */
+  ML_TRAIN_LAYER_TYPE_SUBTRACTION = 31, /**< Subtraction Layer (Since 9.x) */
```
[the remark this answers is not in the document]
Comment (baek2sm) [reply api/nntrainer-api-common.h:67]: thanks

[the remark this answers is not in the document]
Comment (baek2sm) [reply api/nntrainer-api-common.h:67]: > Since 9.0

thanks

**Pull request closed without merging**